src/darwin/Framework/CHIPTests/MTROTAProviderTests.m

```diff
@@ -1579,6 +1581,86 @@ - (void)test008_TestWriteDefaultOTAProviders
     }
 }
 
+#ifdef ENABLE_REAL_OTA_UPDATE_TESTS
+- (void)test009_TestOTAMetrics
+{
+    NSString * otaRawImagePath = [self createRawImageWithVersion:kUpdatedSoftwareVersion_5];
+    NSString * otaImagePath = [otaRawImagePath stringByReplacingOccurrencesOfString:@"raw-image" withString:@"image"];
+
+    // Check whether the ota raw image exists at otaRawImagePath
+    XCTAssertTrue([[NSFileManager defaultManager] fileExistsAtPath:otaRawImagePath]);
+
+    __auto_type * runner = [[MTROTARequestorAppRunner alloc] initWithPayload:kOnboardingPayload1 testcase:self];
+    __auto_type * device = [runner commissionWithNodeID:@(kDeviceId1)];
+
+    dispatch_queue_t queue = dispatch_get_main_queue();
+
+    __auto_type * delegate = [[MTRDeviceTestDelegate alloc] init];
+
+    __auto_type * checker =
+        [[MTROTAProviderTransferChecker alloc] initWithRawImagePath:otaRawImagePath
+                                           otaImageDownloadFilePath:runner.downloadFilePath
+                                                             nodeID:@(kDeviceId1)
+                                                    softwareVersion:kUpdatedSoftwareVersion_5
+                                              softwareVersionString:kUpdatedSoftwareVersionString_5
+                                                  applyUpdateAction:MTROTASoftwareUpdateProviderApplyUpdateActionProceed
+                                                           testcase:self];
+
+    // For some reason we need this to make sure the device in MTROTAImageTransferHandler::InvokeTransferSessionEndCallback
+    // has valid vendor id, product id, etc.
```

Comment:
Isn't the "some reason" just that we need to create a subscription to get this data from the server?

And given that, is this test not racy, in that if OTA wins the race over the subscription (unlikely, I guess, given packet counts), that information would in fact not be available?

As in, shouldn't we wait here until the device cache is primed before we proceed with the rest of the test?